src/bit-fields/src/bit_range.rs

```diff
@@ -0,0 +1,1117 @@
+                if x <= Self::MAX {
+                    let shift = x.get() << START;
+                    // SAFETY: Always safe.
+                    *self.0 = unsafe { <$y>::new_unchecked(shift | (self.0.get() & !BitRangeMut::<$x, START, END>::MASK)) };
+                    Ok(())
+                } else {
+                    Err($crate::CheckedAssignError)
+                }
+            }
+            #[doc = concat!("
+                Conveniance alias for [`Self::MAX`].
+                
+                ```
+                use bit_fields::BitRangeMut;
+                use std::num::", stringify!($y), ";
+                let mut x = ", stringify!($y), "::new(1).unwrap();
+
+                assert_eq!(BitRangeMut::<_,0,1>(&mut x).get_max(),", stringify!($y), "::new(1).unwrap());
+                assert_eq!(BitRangeMut::<_,0,2>(&mut x).get_max(),", stringify!($y), "::new(3).unwrap());
+                assert_eq!(BitRangeMut::<_,0,3>(&mut x).get_max(),", stringify!($y), "::new(7).unwrap());
+                assert_eq!(BitRangeMut::<_,0,4>(&mut x).get_max(),", stringify!($y), "::new(15).unwrap());
+                assert_eq!(BitRangeMut::<_,0,5>(&mut x).get_max(),", stringify!($y), "::new(31).unwrap());
+                assert_eq!(BitRangeMut::<_,0,6>(&mut x).get_max(),", stringify!($y), "::new(63).unwrap());
+                assert_eq!(BitRangeMut::<_,0,7>(&mut x).get_max(),", stringify!($y), "::new(127).unwrap());
+                assert_eq!(BitRangeMut::<_,0,8>(&mut x).get_max(),", stringify!($y), "::new(255).unwrap());
+                ```
+            ")]
+            #[must_use]
+            #[inline]
+            pub const fn get_max(&self) -> $y {
+                Self::MAX
+            }
+            #[doc = concat!("
+                Conveniance alias for [`Self::MASK`].
+                
+                ```
+                use bit_fields::BitRangeMut;
+                use std::num::", stringify!($y), ";
+                let mut x = ", stringify!($y), "::new(1).unwrap();
+
+                assert_eq!(BitRangeMut::<_,0,1>(&mut x).get_mask(),", stringify!($y), "::new(1).unwrap());
+                assert_eq!(BitRangeMut::<_,1,3>(&mut x).get_mask(),", stringify!($y), "::new(6).unwrap());
+                assert_eq!(BitRangeMut::<_,3,7>(&mut x).get_mask(),", stringify!($y), "::new(120).unwrap());
+                assert_eq!(BitRangeMut::<_,7,8>(&mut x).get_mask(),", stringify!($y), "::new(128).unwrap());
+                ```
+            ")]
+            #[must_use]
+            #[inline]
+            pub const fn get_mask(&self) -> $y {
+                Self::MASK
+            }
+        }
+    };
+}
+
+bit_mut_range!(u128, NonZeroU128, mask_u128, max_u128);
+bit_mut_range!(u64, NonZeroU64, mask_u64, max_u64);
+bit_mut_range!(u32, NonZeroU32, mask_u32, max_u32);
+bit_mut_range!(u16, NonZeroU16, mask_u16, max_u16);
+bit_mut_range!(u8, NonZeroU8, mask_u8, max_u8);
+
+// impl From<..> for BitRange<..>
+macro_rules! bit_range_from {
```

Comment:
Same here, we can have 1 macro for both types:
```rust
macro_rules! impl_bit_range_from {
   ($t:ident, $x:ty, $y:ty) => {
      ...
```

Reply:
Yes, made this change.